### DISCUSSION TITLE
`WPS` rule implementation

### DISCUSSION THREAD

**Discussion by Lokejoke** [astral-sh/ruff#14437] [Ideas]
I've implemented some of the wps's opinionated rules.
 I would really appreciate any feedback.

plugin | rule and PR ref | code (docs) | PR ref
-- | -- | -- | --
wps-light | [consecutive-underscores-in-name](https://github.com/Lokejoke/ruff/pull/10) | [WPS116](https://wemake-python-styleguide.readthedocs.io/en/0.19.2/pages/usage/violations/naming.html#wemake_python_styleguide.violations.naming.ConsecutiveUnderscoresInNameViolation) | [(PR ref)](https://github.com/Lokejoke/ruff/pull/10)
wps-light | [reserved-argument-as-variable](https://github.com/Lokejoke/ruff/pull/17) | [WPS117](https://wemake-python-styleguide.readthedocs.io/en/0.19.2/pages/usage/violations/naming.html#wemake_python_styleguide.violations.naming.ReservedArgumentNameViolation) | [(PR ref)](https://github.com/Lokejoke/ruff/pull/17)
wps-light | [unused-variable-accessed](https://github.com/Lokejoke/ruff/pull/8) | [WPS121](https://wemake-python-styleguide.readthedocs.io/en/0.19.2/pages/usage/violations/naming.html#wemake_python_styleguide.violations.naming.UnusedVariableIsUsedViolation) | [(PR ref)](https://github.com/Lokejoke/ruff/pull/8)
wps-light | [underscores-in-number](https://github.com/Lokejoke/ruff/pull/12) | [WPS303](https://wemake-python-styleguide.readthedocs.io/en/0.19.2/pages/usage/violations/consistency.html#wemake_python_styleguide.violations.consistency.UnderscoredNumberViolation) | [(PR ref)](https://github.com/Lokejoke/ruff/pull/12)
wps-light | [assignment-to-subscript-slice](https://github.com/Lokejoke/ruff/pull/14) | [WPS362](https://wemake-python-styleguide.readthedocs.io/en/0.19.2/pages/usage/violations/consistency.html#wemake_python_styleguide.violations.consistency.AssignToSliceViolation) | [(PR ref)](https://github.com/Lokejoke/ruff/pull/14)
wps-light | [list-multiplication](https://github.com/Lokejoke/ruff/pull/5) | [WPS435](https://wemake-python-styleguide.readthedocs.io/en/0.19.2/pages/usage/violations/best_practices.html#wemake_python_styleguide.violations.best_practices.ListMultiplyViolation) | [(PR ref)](https://github.com/Lokejoke/ruff/pull/5)
wps-light | [getter-without-return-value](https://github.com/Lokejoke/ruff/pull/4) | [WPS463](https://wemake-python-styleguide.readthedocs.io/en/0.19.2/pages/usage/violations/best_practices.html#wemake_python_styleguide.violations.best_practices.GetterWithoutReturnViolation) | [(PR ref)](https://github.com/Lokejoke/ruff/pull/4)
wps-light | [in-compare-with-single-item-container](https://github.com/Lokejoke/ruff/pull/11) | [WPS525](https://wemake-python-styleguide.readthedocs.io/en/0.19.2/pages/usage/violations/refactoring.html#wemake_python_styleguide.violations.refactoring.WrongInCompareTypeViolation) | [(PR ref)](https://github.com/Lokejoke/ruff/pull/11)
wps-light | [wrong-class-body-content](https://github.com/Lokejoke/ruff/pull/16) | [WPS604](https://wemake-python-styleguide.readthedocs.io/en/0.19.2/pages/usage/violations/oop.html#wemake_python_styleguide.violations.oop.WrongClassBodyContentViolation) | [(PR ref)](https://github.com/Lokejoke/ruff/pull/16)

* would any fit inside ruff?

**Comment by MichaReiser** [comment]
Oh wow, nice. https://github.com/astral-sh/ruff/issues/3845 is the related tracking issue

The reason we haven't added any wemake styleguide rules yet is that many of them are very opinionated and sometimes overlap or even conflict with the formatter. But that doesn't apply to all the rules. 

After doing a quick scan over the listed rules:

* list-multiplication: Related to https://github.com/astral-sh/ruff/issues/10505
* https://github.com/Lokejoke/ruff/pull/11 seems reasoanble, but it's unclear how common this is.
* wrong-class-body-content: Seems reasonable 
* https://github.com/Lokejoke/ruff/pull/8 looks useful


The other rules seem a bit on the too opinionated end (IMO)

**Reply by Lokejoke** [reply]
I agree. I'll reformat those four as separate Ruff rules and submit an individual PR for each. This way, we can easily discard any that turn out to be unnecessary. (I wanted to avoid spamming the PR section)

Does that sound good?

**Reply by MichaReiser** [reply]
That sounds good. It may take some time for us to get to review the new rules because we're currently busy with prepping for the next rough release. Thanks for asking before creating that many PRs :)

One thing we'll have to decide that's relevant for all rules is whether we want to add a wemake styleguide group or add the rules under the ruff prefix.
